infrastructure/code/code.go

```diff
@@ -134,42 +182,18 @@ func (sc *Scanner) Scan(ctx context.Context, _ string, folderPath string) (issue
 	sc.runningScans[folderPath] = scanStatus
 	sc.scanStatusMutex.Unlock()
 	if waitForPreviousScan {
-		<-previousScanStatus.finished // Block here until previous scan is finished
+		<-previousScanStatus.finished
 
-		// Setting isPending = false allows for future scans to wait for the current
-		// scan to finish, instead of returning immediately
 		sc.scanStatusMutex.Lock()
 		scanStatus.isPending = false
 		sc.scanStatusMutex.Unlock()
 	}
-	defer func() {
+	return false, func() {
```

Comment:
Why is the clean up for the waiting function exposed to the outside? Shouldn't it stay in the function? Is there a better solution for not returning the cleanup?